src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -1125,34 +952,26 @@ private static unsafe void WriteThreeLowOrderBytes(byte* destination, int value)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static unsafe int GetIndexOfFirstWhitespace(byte* src)
+        private static bool IsWhitespace(int value)
         {
-            for (int i = 0; i < 4; i++)
+            if (Environment.Is64BitProcess)
             {
-                if (IsByteWhitespace(src[i]))
-                {
-                    return i;
-                }
-            }
+                const ulong MagicConstant = 0xC800010000000000UL;
```

Comment:
At the moment we know how this magic constant is created. But do we know it still next year? I'm rather sure I'll forget it. So please but a comment similar [to this one](https://github.com/gfoidl/Base64/blob/a4968adf3d9b68390797a1290a6ff1e9f61cf7e3/source/gfoidl.Base64/Internal/Operations/CommonOperation.cs#L15-L31) here (at least the link to the _HexConverter_).